lib/plug/adapters/test/conn.ex

```diff
@@ -148,9 +148,16 @@ defmodule Plug.Adapters.Test.Conn do
   defp body_or_params(params, query, headers, method)
        when is_map(params) and method in ["GET", "HEAD"] do
     params = stringify_params(params, &to_string/1)
-    params = Map.merge(Plug.Conn.Query.decode(query), params)
 
-    {"", nil, params, headers}
+    query = Plug.Conn.Query.decode(query)
+    params = Map.merge(query, params)
+
+    query =
+      params
+      |> Map.merge(query)
```

Comment:
Is there a reason why we reverse the merge order here? Shouldn't it be the same as the encoded params?

Reply:
The reason is with this test case:

```elixir
    conn = conn(:get, "/?foo=bar", %{foo: "baz"})
    assert conn.body_params == %Plug.Conn.Unfetched{aspect: :body_params}
    assert conn.query_string == "foo=bar"
    assert conn.params == %{"foo" => "baz"}
```

In that case, do we expect `query_string` to be `foo=bar` or `foo=baz` when we have duplicate keys?

With the current implementation we ensure that the query string takes precedence (which I thought makes more sense), if we encode it with the with the original order (or rather, encode `params` altogether since they're the same thing), the param takes precedence.

Basically, `body_params` would now then take the third parameter as precedence, but `query_string` would take the original uri string itself as precedence. It's illustrated in the test case below:

```elixir
    conn = conn(:post, "/?foo=bar", %{foo: "baz", answer: 42})
    assert conn.body_params == %{"foo" => "baz", "answer" => 42}
    assert conn.query_string == "foo=bar"
    assert conn.params == %{"foo" => "baz", "answer" => 42}
```

As you can see, `query_string` follows the URI, but `body_params` follow the params. Would this be desired?

Reply:
You are correct on all points. Thank you.